File: src/main/java/upbrella/be/store/dto/request/CreateClassificationRequest.java
package upbrella.be.store.dto.request;

import lombok.*;
import org.hibernate.validator.constraints.Range;

import javax.validation.constraints.NotBlank;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class CreateClassificationRequest {

    @NotBlank
    private String name;
    @Range(min = -90, max = 90)
    private double latitude;
    @Range(min = -180, max = 180)
    private double longitude;
}

File: src/main/kotlin/upbrella/be/store/dto/request/CreateClassificationRequest.kt
package upbrella.be.store.dto.request

import org.hibernate.validator.constraints.Range
import javax.validation.constraints.NotBlank

data class CreateClassificationRequest(

    @field:NotBlank
    val name: String,

    @field:Range(min = -90, max = 90)
    val latitude: Double,

    @field:Range(min = -180, max = 180)
    val longitude: Double
)
